go/fn/object.go

```diff
@@ -169,20 +169,41 @@ func (o *SubObject) NestedSliceOrDie(fields ...string) SliceSubObjects {
 	return val
 }
 
-// NestedInt64 returns the int64 value, if the field exist and a potential error.
-func (o *SubObject) NestedMap(fields ...string) (*SubObject, bool, error) {
```

Comment:
So `NestedMap` is completely gone?

Reply:
Right, users should use `GetMap` or `UpsertMap` and expect a SubObject return value
One key change of this PR is to clarify that users should not pass interface or get an interface returned when using the `NestedXXXField` `SetNesetdXXXField`, because it will be very error-prone and loses the SubObject/KubeObject access.